### PR TITLE
Provide additional attributes in the API representation of documents and mails

### PR DESCRIPTION
**Pull request by phgross**
For #5905 

We also wondered during the Scrum meeting if it would make sense to namespace the additional attributes or use the `metadata_fields` behavior. I decided against it for the following reasons:
 - The document representation already contains "ready-only" metadata which is not part of a schema on the top level
 - Non schema keys are ignored by the restapi on Patch and Post requests
 - `metadata_fields` is currently not supported for the representation of a context

## Checkliste

_Zutreffendes soll angehakt stehengelassen werden._
- [x] Gibt es neue Funktionalität mit einem `Dokument`? Funktioniert das auch mit einem `Mail`?
- [x] Changelog-Eintrag vorhanden/nötig?
- [ ] Aktualisierung Dokumentation vorhanden/nötig? Nein, ich bin der Meinung das die zusätzlichen Werte nicht zusätzlich erwähnt werden müssen.

### CODE REVIEW

File: opengever/api/tests/test_document.py [hunk 1]
```diff
@@ -48,6 +48,40 @@ def test_document_serialization_contains_file_extension(self, browser):
...
+        self.assertEqual(u'2016', browser.json['containing_subdossier'])
+        self.assertFalse(browser.json['trashed'])
+        self.assertFalse(browser.json['is_shadow_document'])
+        self.assertFalse(0, browser.json['current_version_id'])
+
+    @browsing
+    def test_additional_metadata_for_mails(self, browser):
+        self.login(self.regular_user, browser)
+
+        browser.open(self.subdocument, headers={'Accept': 'application/json'})
```
Comment (njohner) [inline opengever/api/tests/test_document.py]: `self.mail` probably?

File: opengever/document/tests/test_base.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    def test_is_trashed(self):
+        self.login(self.regular_user)
+        self.assertFalse(self.base_document.is_trashed)
+
+        self.trash_documents(self.base_document)
+        self.assertTrue(self.base_document.is_trashed)
+
+
+class TestBaseDocumentMails(TestBaseDocument):
```
Comment (njohner) [inline opengever/document/tests/test_base.py]: That's nice!

**Comment by phgross** [comment]
The `containing_dossier` and `containing_subdossier` indexers a registered and implemented for every IDexterityContent, so the indexers work also for documents and mails. As discussed it does not make much sense to register a specific one for documents which does the same as the general one.

**Comment by njohner** [comment]
Sorry, didn't realise I should have another look...

**Pull request merged**